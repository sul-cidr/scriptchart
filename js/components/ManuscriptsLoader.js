import React from "react";

/* ManuscriptsLoader - Renders the list of available manuscripts in the
 * ManuscriptsMenu form component, and passes their selection/deselection
 * events up the form component chain.
 * It doesn't actually load the manuscripts list from an external source;
 * this data has already been queried via the REST API by the App component.
 */

class ManuscriptsLoader extends React.Component {
  constructor(props) {
    super(props);
    this.state = { selectedShelfmarks: [] };

    this.handleSelect = this.handleSelect.bind(this);
  }

  handleSelect(event) {
    let options = event.target.options;
    let name = event.target.name;
    let value = [];
    let selectedShelfmarks = [];
    for (let i = 0, l = options.length; i < l; i++) {
      if (options[i].selected) {
        value.push(options[i].value);
        selectedShelfmarks.push(options[i].value);
      }
    }
    this.setState({ selectedShelfmarks });

    this.props.handleSelect(name, value);
  }

  render() {
    let manuscriptSelectors = this.props.manuscripts.map(ms => {
      return (
        <option key={ms.id} value={ms.shelfmark}>
          {ms.shelfmark}
        </option>
      );
    });

    return (
      <select
        type="string"
        name="selectedShelfmarks"
        value={this.state.selectedShelfmarks}
        onChange={this.handleSelect}
        multiple={true}
      >
        {manuscriptSelectors}
      </select>
    );
  }
}

export default ManuscriptsLoader;
