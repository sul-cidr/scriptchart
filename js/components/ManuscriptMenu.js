import React from "react";

/* ManuscriptMenu - A sub-component of the ManuscriptForm; provides
 * a multi-select list of the available manuscripts (encased in
 * the ManuscriptsLoader component) and options for sorting and
 * multi-selecting the list elements.
 */

import ManuscriptsLoader from "./ManuscriptsLoader";

class ManuscriptMenu extends React.Component {
  constructor(props) {
    super(props);

    this.state = { orderByShelfmark: true };

    this.changeSort = this.changeSort.bind(this);
    this.manuscriptsSelect = this.manuscriptsSelect.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
  }

  handleSelect(event) {
    let options = event.target.options;
    let name = event.target.name;
    let value = [];
    let selectedShelfmarks = [];
    for (let i = 0, l = options.length; i < l; i++) {
      if (options[i].selected) {
        value.push(options[i].value);
        selectedShelfmarks.push(options[i].value);
      }
    }

    this.props.handleSelect(name, value);
  }

  // Currently this is a simple all/none toggle
  manuscriptsSelect(event) {
    let selectedShelfmarks = JSON.parse(
      JSON.stringify(this.props.selectedShelfmarks)
    );
    for (let ms of this.props.manuscripts) {
      let sm = ms.shelfmark;
      if (//which == "All" &&
        selectedShelfmarks.indexOf(sm) < 0) {
        selectedShelfmarks.push(sm);
      } else {
        selectedShelfmarks = [];
        break;
      }
    }
    this.props.handleSelect("selectedShelfmarks", selectedShelfmarks);
  }

  changeSort(event) {
    const value = event.target.value;

    if (value == "shelfmark") {
      this.setState({ orderByShelfmark: true });
    } else {
      // value == "date"
      this.setState({ orderByShelfmark: false });
    }
    this.props.sortManuscripts(value);
  }

  render() {
    return (
      <div className={"field"}>
        <div className={"control"} style={{ marginBottom: 5 }}>
          <label className={"control"}>Select manuscripts: </label>
          <span className="button is-small" onClick={this.manuscriptsSelect}>
            All/None
          </span>
        </div>
        <div className={"select is-multiple"}>
          <ManuscriptsLoader
            handleSelect={this.handleSelect}
            manuscripts={this.props.manuscripts}
            selectedShelfmarks={this.props.selectedShelfmarks}
          />
        </div>
        <div className={"control"}>
          <p>Order manuscripts by</p>
          <label className={"radio"}>
            <input
              type="radio"
              value="shelfmark"
              onChange={this.changeSort}
              checked={this.state.orderByShelfmark}
            />{" "}
            Shelfmark |
          </label>
          <label className={"radio"}>
            <input
              type="radio"
              value="date"
              onChange={this.changeSort}
              checked={!this.state.orderByShelfmark}
            />{" "}
            Date
          </label>
        </div>
      </div>
    );
  }
}

export default ManuscriptMenu;
