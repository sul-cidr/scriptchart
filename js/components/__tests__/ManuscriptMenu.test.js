/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import ManuscriptMenu from "../ManuscriptMenu";
import { defaultManuscripts } from "../ManuscriptsLoader";

describe("ManuscriptMenu test", () => {
  it("ManuscriptMenu should match snapshot", () => {
    const component = renderer.create(
      <ManuscriptMenu manuscripts={defaultManuscripts} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
