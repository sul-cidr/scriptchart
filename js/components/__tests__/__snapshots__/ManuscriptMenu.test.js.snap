// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManuscriptMenu test ManuscriptMenu should match snapshot 1`] = `
<div
  className="field"
>
  <label
    className="control"
  >
    Select Manuscripts:
  </label>
  <div
    className="control"
  >
    <div
      className="select is-multiple"
    >
      <select
        multiple={true}
        name="selectedShelfmarks"
        onChange={[Function]}
        type="string"
        value={Array []}
      >
        <option
          value="Vat. Syr. 092"
        >
          Vat. Syr. 092
        </option>
        <option
          value="Vat. Syr. 112"
        >
          Vat. Syr. 112
        </option>
        <option
          value="Vat. Syr. 140"
        >
          Vat. Syr. 140
        </option>
        <option
          value="BL. Add. 17224"
        >
          BL. Add. 17224
        </option>
        <option
          value="BL. Add. 12144"
        >
          BL. Add. 12144
        </option>
        <option
          value="BL. Add. 12134"
        >
          BL. Add. 12134
        </option>
        <option
          value="BL. Add. 14445"
        >
          BL. Add. 14445
        </option>
        <option
          value="BL. Add. 12146"
        >
          BL. Add. 12146
        </option>
        <option
          value="BL. Or. 8732"
        >
          BL. Or. 8732
        </option>
        <option
          value="BL. Or. 8606"
        >
          BL. Or. 8606
        </option>
        <option
          value="BL. Add. 17102"
        >
          BL. Add. 17102
        </option>
        <option
          value="BL. Add. 17107"
        >
          BL. Add. 17107
        </option>
        <option
          value="BL. Add. 12175"
        >
          BL. Add. 12175
        </option>
      </select>
    </div>
  </div>
  <div
    className="control sort-option"
  >
    <p>
      Order manuscripts by:
    </p>
    <label
      className="radio"
    >
      Shelfmark:
      <input
        defaultChecked={true}
        name="Shelfmark"
        type="radio"
      />
    </label>
    <label
      className="radio"
    >
      Date:
      <input
        name="Date"
        type="radio"
      />
    </label>
  </div>
</div>
`;
